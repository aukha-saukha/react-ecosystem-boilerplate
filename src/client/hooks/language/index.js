// @flow strict

import { LanguageProvider, useLanguage } from './language.react';

export { LanguageProvider, useLanguage };
