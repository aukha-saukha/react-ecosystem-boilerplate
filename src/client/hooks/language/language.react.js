// @flow strict

import * as React from 'react';

import { DEFAULT_LOCALE } from '../../../data/constants/app/config';

import getUserLocale from '../../utilities/user-locale';

import type { LanguageProviderParametersType } from './language.type';

// TODO: Fix flow error here.
const LanguageContext: React.Context<
  [string, (((string) => string) | string) => void]
  // flow-disable-line
> = React.createContext();

LanguageContext.displayName = 'LanguageContext';

function LanguageProvider(props: LanguageProviderParametersType) {
  const { children } = props;
  const [language, setLanguage] = React.useState<string>(DEFAULT_LOCALE);

  React.useEffect(() => {
    async function getUserLocaleFromLocaleUtility() {
      const userLanguage = await getUserLocale();
      setLanguage(userLanguage);
    }

    getUserLocaleFromLocaleUtility();
  }, [language]);

  const providerValue = React.useMemo(() => [language, setLanguage], [language]);
  return <LanguageContext.Provider value={providerValue}>{children}</LanguageContext.Provider>;
}

function useLanguage() {
  const context = React.useContext(LanguageContext);

  if (typeof context === 'undefined') {
    throw new Error(`useLanguage must be used within a LanguageProvider`);
  }

  const [language, setLanguage] = context;

  return {
    language,
    setLanguage,
  };
}

export { LanguageProvider, useLanguage };
