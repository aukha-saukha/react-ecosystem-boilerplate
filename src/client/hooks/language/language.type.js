// @flow strict

import type { Node } from 'react';

export type LanguageProviderParametersType = {|
  children: Node,
|};
