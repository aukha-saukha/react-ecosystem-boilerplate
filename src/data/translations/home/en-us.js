/**
 * Copyright (c) 2020-present Aukha Saukha Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

// @flow strict

const homePageMessage = 'Home page';

export { homePageMessage };
