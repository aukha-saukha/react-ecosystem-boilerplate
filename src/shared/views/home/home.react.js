import React from 'react';

function Home() {
  return (
    <main className="font-size-16" data-test="main">
      Home page
    </main>
  );
}

export default Home;
