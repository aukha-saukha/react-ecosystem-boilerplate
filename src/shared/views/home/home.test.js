/**
 * Copyright (c) 2020-present Aukha Saukha Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { shallow } from 'enzyme';
import React from 'react';

import { LanguageProvider } from '@hooks/language';

import { Home } from './home.react';

describe('Home component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <LanguageProvider>
        <Home />
      </LanguageProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
