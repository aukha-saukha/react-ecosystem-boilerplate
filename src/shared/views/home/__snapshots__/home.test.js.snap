// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders correctly 1`] = `
<main
  className="font-size-16"
>
  Home page
</main>
`;
